docs/dev/concepts.rst

```diff
@@ -158,16 +121,13 @@ distributed by the service.
 This helps prevent compromise of a single account from compromising the entire
 service, since two accounts need to be compromised to publish a recipe.
 
-Action Sandbox
-^^^^^^^^^^^^^^
-Actions are executed within a JavaScript sandbox by the runtime. The sandbox
-limits the access of the JavaScript to prevent it from modifying Firefox in ways
-that haven't been reviewed and approved beforehand.
-
-To perform actions that JavaScript normally can't (such as displaying a
-Heartbeat survey), the action in the sandbox is passed a
-:doc:`driver object </dev/driver>`, which contains methods that can modify the
-client or trigger other privileged behavior.
+Static Instructions
+^^^^^^^^^^^^^^^^^^^
+Recipe data is provided in either a static format (JSON) or an executable
```

Comment:
Here is the second place where you used `static` that made me think something was amiss. I guess you are using the word `static` here to indicate that the instructions are not in the form of JavaScript code. I am not really sure this is an important distinction (I have a friend who likes to point out that not only code is data, but also data is code) but if so you might simply say that recipe information is not Turing-complete.